pallets/clusters/src/lib.rs

```diff
@@ -318,13 +375,57 @@ pub mod pallet {
 				let role_hash = blake2_256(&[role.clone(), cluster_name.clone()].concat());
 				<Roles<T>>::mutate(&role_hash, |role| {
 					let role = role.as_mut().unwrap();
-					role.members.push(member.clone());
+					role.members.push(member_hash.clone());
 				});
 			}
 
 			Ok(())
 		}
 
+		/// Edit the data of a **Member** in an existing cluster.
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1).ref_time())]
+		pub fn edit_member(
+			origin: OriginFor<T>,
+			cluster_hash: Hash256,
+			member_hash: Hash256,
+			new_member_data: Option<Vec<u8>>,
```

Comment:
we are allowing arbitrary lengths of data to be stored inside the actual chain state here.. this is not good.
I just realised we have many of those `Vec<u8>` they all need to be replaced with BoundedVec, whenever the data is STORED in the live state. (See protos upload, data IS NOT STORED anywhere that's why unbound is ok)
Let's make a story about this asap as we are increasing block size and capabilities soon, and this refactor needs to come first.